tls/client-tls-pkcallback.c

```diff
@@ -182,6 +193,119 @@ static int myEccSign(WOLFSSL* ssl, const byte* in, word32 inSz,
 
     return ret;
 }
+#endif
+
+#ifndef NO_RSA
+static int myRsaSign(WOLFSSL* ssl, const byte* in, word32 inSz,
+        byte* out, word32* outSz, const byte* key, word32 keySz, void* ctx)
+{
+    int     ret;
+    word32  idx = 0;
+    byte*   keyBuf = (byte*)key;
+    PkCbInfo* cbInfo = (PkCbInfo*)ctx;
+
+    (void)ssl;
+    (void)cbInfo;
+
+    printf("PK RSA Sign: inSz %u, keySz %u\n", inSz, keySz);
+
+#ifdef WOLFSSL_ASYNC_CRYPT
+    if (cbInfo->state == 0) {
+        cbInfo->state++;
+        printf("PK ECC Sign: Async Simulate\n");
+        return WC_PENDING_E;
+    }
+#endif
+
+    ret = load_key_file(cbInfo->keyFile, &keyBuf, &keySz);
+    if (ret != 0)
+        return ret;
+
+    ret = wc_InitRsaKey(&cbInfo->keyRsa, NULL);
+    if (ret == 0) {
+        ret = wc_RsaPrivateKeyDecode(keyBuf, &idx, &cbInfo->keyRsa, keySz);
+        if (ret == 0) {
+            WC_RNG *rng = wolfSSL_GetRNG(ssl);
+            ret = wc_RsaSSL_Sign(in, inSz, out, *outSz, &cbInfo->keyRsa, rng);
+        }
+        if (ret > 0) {  /* save and convert to 0 success */
+            *outSz = ret;
+            ret = 0;
+        }
+        wc_FreeRsaKey(&cbInfo->keyRsa);
+    }
+    free(keyBuf);
+#ifdef WOLFSSL_ASYNC_CRYPT
+    cbInfo->state = 0;
+#endif
+
+    printf("PK RSA Sign: ret %d, outSz %u\n", ret, *outSz);
+
+    return ret;
+}
+
+#ifdef WC_RSA_PSS
+static int myRsaPssSign(WOLFSSL* ssl, const byte* in, word32 inSz,
+        byte* out, word32* outSz, int hash, int mgf, const byte* key,
+        word32 keySz, void* ctx)
+{
+    enum wc_HashType hashType = WC_HASH_TYPE_NONE;
+    int              ret;
+    word32           idx = 0;
+    byte*            keyBuf = (byte*)key;
+    PkCbInfo* cbInfo = (PkCbInfo*)ctx;
+
+    (void)ssl;
+    (void)cbInfo;
+
+    printf("PK RSA PSS Sign: inSz %u, hash %d, mgf %d, keySz %u\n",
+        inSz, hash, mgf, keySz);
+
+    ret = load_key_file(cbInfo->keyFile, &keyBuf, &keySz);
+    if (ret != 0)
+        return ret;
+
+    switch (hash) {
+#ifndef NO_SHA256
+        case SHA256h:
+            hashType = WC_HASH_TYPE_SHA256;
+            break;
+#endif
+#ifdef WOLFSSL_SHA384
+        case SHA384h:
+            hashType = WC_HASH_TYPE_SHA384;
+            break;
+#endif
+#ifdef WOLFSSL_SHA512
+        case SHA512h:
+            hashType = WC_HASH_TYPE_SHA512;
+            break;
+#endif
+    }
```

Comment:
Is it possible that all of these are disabled and then causes an empty switch statement? Perhaps add `default:` ?